modules/datastore/src/Service/DatastoreQuery.php

```diff
@@ -14,9 +14,14 @@ class DatastoreQuery extends RootedJsonData {
    *
    * @param string $json
    *   JSON query string from API payload.
+   * @param int $rows_limit
+   *   Maxmimum rows of data to return.
    */
-  public function __construct(string $json) {
+  public function __construct(string $json, int $rows_limit = 500) {
```

Comment:
See https://github.com/GetDKAN/dkan/pull/3689#pullrequestreview-774168989.
```suggestion
  public function __construct(string $json, int $rows_limit) {
```